### PR TITLE
Add React Client Package

### PR DESCRIPTION
**Pull request by frankfka**
Extracted `@vertex-protocol/web-data` from the web monorepo into this SDK repo. Ideally, I would have liked to publish the package directly from the web monorepo, but was having issues with build. This seems to work fine though!

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -3,5 +3,5 @@
   "useNx": true,
   "useWorkspaces": true,
   "npmClient": "yarn",
-  "version": "0.34.0"
+  "version": "0.34.2"
```
Comment (frankfka) [inline lerna.json:6]: Had to publish for testing

File: packages/web-data/package.json
Comment (frankfka) [inline packages/web-data/package.json:1]: This file is important to take a look at

File: packages/web-data/package.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  "scripts": {
+    "clean": "rm -rf dist && rm -f tsconfig.tsbuildinfo",
+    "build": "tsc -b",
+    "dev": "tsc -w"
+  },
+  "dependencies": {
+    "d3-format": "^3.1.0"
+  },
+  "peerDependencies": {
+    "@tanstack/react-query": "^5",
```
Comment (frankfka) [inline packages/web-data/package.json:35]: More permissive versioning here for consumers. `devDependencies` is what's required for local dev on the SDK

File: packages/web-data/src/context/evm/CoreEVMContextProvider.tsx
Comment (frankfka) [inline packages/web-data/src/context/evm/CoreEVMContextProvider.tsx:1]: These other files are copied over, so not sure if you need to check

File: packages/web-data/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# `@vertex-protocol/web-data`
```
Comment (yruej301) [inline packages/web-data/README.md:1]: a really nice to have but would be nice to show how a frontend can use this package and maybe show an example, etc

File: packages/web-data/src/hooks/useIsChainType.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { useEVMContext } from '../context';
+
+export function useIsChainType() {
```
Comment (yruej301) [inline packages/web-data/src/hooks/useIsChainType.ts:3]: is it missing `isMantle`?

**Review by yruej301 — COMMENTED** [review]
left some minor comments, i think we should rename the package to something like `react-client`

File: packages/web-data/README.md (hunk 3 above)
Comment (frankfka) [reply packages/web-data/README.md:1]: Yeah - this is one thing we should look into a lot more. I added a [task](https://www.notion.so/vertexprotocol/Public-Documentation-for-using-vertex-protocol-react-client-aa226a97ad2547aaa60065149cd96941?pvs=4). For now, this is extracted so we can split out Blitz

File: packages/web-data/package.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  "publishConfig": {
+    "access": "public"
+  },
+  "files": [
+    "dist",
+    "src"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/vertex-protocol/vertex-sdk",
```
[the remark this answers is not in the document]
Comment (frankfka) [reply packages/web-data/package.json:23]: Going to find & replace all of these

File: packages/web-data/src/hooks/useIsChainType.ts (hunk 4 above)
Comment (frankfka) [reply packages/web-data/src/hooks/useIsChainType.ts:3]: Yeah, guess we never ended up using this hook :P - will add

**Review by yruej301 — APPROVED** [review]
looks great, just one tiny thing to fix but otherwise lgtm!

**Pull request merged**